### PR TITLE
Introducing a new log level: STEP

### PR DESCRIPTION
**Pull request by michaelk-stratoscale**
Please review
@eyal-stratoscale @rom-stratoscale @arie-stratoscale

### CODE REVIEW

File: py/strato/common/log/coloringformatter.py [hunk 1]
```diff
@@ -23,6 +27,10 @@ def format(self, record):
             record.startColor = self._GREEN
         elif record.levelno == _WARNING:
             record.startColor = self._YELLOW
+        elif record.levelno == _STEP:
+            record.startColor = self._LIGHT_GREEN
+            record.levelname = "STEP [%d]" % self.STEP_COUNT
+            self.STEP_COUNT += 1
```
Comment (rom-stratoscale) [inline py/strato/common/log/coloringformatter.py]: what will happen if you run with multiple threads? how the count will behave? I thinks you need a lock here

Comment (rom-stratoscale) [inline py/strato/common/log/coloringformatter.py]: Or it's a different logger?

**Comment by alon-stratoscale** [comment]
Good idea :+1: 
Ship it.

File: py/strato/common/log/coloringformatter.py (hunk 1 above)
Comment (michaelk-stratoscale) [inline py/strato/common/log/coloringformatter.py]: I think practically this is not a big issue. ColoringFormatter is used only for StreamHandler in logging library. Potentially there could be a race here where few competing threads will log.step simultaniously... but the only consequence would be two STEP prints with same step number.

Comment (rom-stratoscale) [inline py/strato/common/log/coloringformatter.py]: Its not a big issue. But if we aim for correction....

**Comment by michaelk-stratoscale** [comment]
Merged

**Pull request closed without merging**